src/System.Text.Json/tests/WritableJsonApiTests.cs

```diff
@@ -624,6 +966,324 @@ public static void TestAquiringAllProperties()
+        {
+            // Retrieve Json from network, should not throw any exceptions:
+            JsonNode receivedEmployeeData = JsonElement.DeepCopy(Mailbox.RetrieveMutableEmployeeData());
+            if (receivedEmployeeData is JsonObject employee)
+            {
+                employee["name"] = new JsonString("Bob");
+                Mailbox.SendEmployeeData(employee.AsJsonElement());
+            }
+        }
+
+        /// <summary>
+        /// Transforming JsonDocument to JsonNode and vice versa
+        /// </summary>
+        [Fact]
+        public static void TestTransformingToFromJsonDocument()
+        {
+            string jsonString = @"
+            {
+                ""employee1"" : 
+                {
+                    ""name"" : ""Ann"",
+                    ""surname"" : ""Predictable"",
+                    ""age"" : 30,                
+                },
+                ""employee2"" : 
+                {
+                    ""name"" : ""Zoe"",
+                    ""surname"" : ""Coder"",
+                    ""age"" : 24,                
+                }
+            }";
+
+            using (JsonDocument employeesToSend = JsonDocument.Parse(jsonString))
+            {
+                // regular send:
+                Mailbox.SendAllEmployeesData(employeesToSend.RootElement);
+
+                // modified elements send:
+                JsonNode modifiableDocument = JsonElement.DeepCopy(employeesToSend);
+                var employees = modifiableDocument as JsonObject;
+                Assert.Equal(2, employees.Count());
+
+                employees.Add(EmployeesDatabase.GetNextEmployee());
+                Assert.Equal(3, employees.Count());
+
+                Mailbox.SendAllEmployeesData(employees.AsJsonElement());
+            }
+        }
+
+        /// <summary>
+        /// Parsing right to JsonNode if user knows data will be modified
+        /// </summary>
+        [Fact]
+        public static void TestParsingToJsonNode()
+        {
+            string jsonString = @"
+            {
+                ""employee1"" : 
+                {
+                    ""name"" : ""Ann"",
+                    ""surname"" : ""Predictable"",
+                    ""age"" : 30,                
+                },
+                ""employee2"" : 
+                {
+                    ""name"" : ""Zoe"",
+                    ""surname"" : ""Coder"",
+                    ""age"" : 24,                
+                }
+            }";
+
+            JsonObject employees = JsonNode.Parse(jsonString) as JsonObject;
+            employees.Add(EmployeesDatabase.GetNextEmployee());
+            Mailbox.SendAllEmployeesData(employees.AsJsonElement());
+
+            Assert.Equal(2, employees.Count());
+            Assert.True(employees.ContainsProperty("employee1"));
+            Assert.True(employees.ContainsProperty("employee2"));
+
+            JsonObject employee2 = employees.GetObjectProperty("employee2");
```

Comment:
`GetObjectProperty` -> `GetProperty`?

Reply:
Then I would need to cast it.

Reply:
Oh I see what you mean, so when you say GetObjectProperty you mean that the property that you are getting is itself an object. This isn't very intuitive, since it is a method inside JsonObject, to me the `Object` on GetObjectProperty is `this`.

Reply:
`GetPropertyAsJsonObject`?

Reply:
Or `GetJsonObjectProperty`?